spec/src/main/asciidoc/config.asciidoc

```diff
@@ -0,0 +1,126 @@
+----
+mp.<spec-name>
+e.g. The Messaging Specification has an @Outgoing annotation:
+mp.messaging.outgoing.value
+----
+
+[[qualifyingClassNames]]
+==== Properties Names Representing User Class names Should Be Package Qualified
+
+There are places where an annotation class name appears or a specification sourced class name (e.g. Concurrency) appears in configuration property names, these class names are not package qualified: 
+----
+//For example: 
+   Retry/maxRetries=30
+//or should this be?
+   mp.ft.retry.maxRetries=30
+----
+This is probably the right choice from a usability point of view. 
+This is particularly not a problem where the scope of the configuration property identifies a particular Java file (which will have resolved the package name via an import).
+It could become more of a problem for service wide defaults if more than one MicroProfile service
+has a class with the same name. 
+The result is that we should have exclusive class names in MicroProfile for classes that might need configuration override in the process environment. 
+The 'first' specification that owns a class name should own that name exclusively. 
+
+Where user class names appear, these are expected to be package qualified.
+
+[[overloadedMethods]]
+==== Properties That Relate To Overloaded Methods
+
+There are places in the configuration where a method name is used. 
+Accounting for methods with the same name but different signatures
+can be done using indexing. 
+The "1" in the Property name below refers to the occurrence of the 'method' named 
+----
+`org.eclipse.microprofile.example.ExampleBean/setCompletableFuture/1/ManagedExecutorCon fig/maxAsync=5`
+----
+
+[[scopedOverriding]]
+==== Properties Overriding a Particular Scope.
+
+Various specifications consider overriding, either at particular annotation location, a particular class or
+service wide (treated as global). Overriding at different scopes should occur as follows: 
+
+The set of possible scopes is:
+
+- class - `<dotted-fully-package-qualified-classname>/<annotation-classname>/<parameter>`
+- package-prefix `<package-prefix>.*` ( `*` is not a valid ENV variable char)
+- package - `<dotted-fully-qualified-package>/<annotation-classname>/<parameter>` (no user class or method name)
+- archive - not specified yet
+- service - [`mp.service-shortname.`]`<annotation-or-classname>/<parameter>`
+- globally - same as service
+- named - Where an annotation occurrence is named - not specified yet.
+
+[[overridingPrecedence]]
+==== Precedence of Configuration Scope Overlap
+
+Configuration which is more specific overrides configuration where the scope is more general.
+
+For example, where multiple configuration properties are present, the property `<classname>/<methodname>/<annotation>/<parameter>` takes precedence over `<classname>/<annotation>/<parameter>`, which would take precedence over `<annotation>/<parameter>`.
+
+[[indexingOverrides]]
+==== Indexing Annotation Occurrences (from Concurrency)
```

Comment:
MP Concurrency has been renamed to MP Context Propagation, although it no longer contains the ManagedExecutorConfig/ThreadContextConfig annotations as indicated in one of my earlier comments.

Reply:
OK understood  -  I will fix.